Dockerfile

```diff
@@ -146,21 +147,20 @@ RUN useradd -m -s /bin/bash -G users smriprep
 WORKDIR /home/smriprep
 ENV HOME="/home/smriprep"
 
-# Precaching atlases
-RUN pip install --no-cache-dir "templateflow<0.4.0a0,>=0.3.0" && \
-    python -c "from templateflow import api as tfapi; \
-               tfapi.get('MNI152Lin|MNI152NLin2009cAsym|OASIS30ANTs', suffix='T1w'); \
-               tfapi.get('MNI152Lin|MNI152NLin2009cAsym|OASIS30ANTs', desc='brain', suffix='mask'); \
-               tfapi.get('OASIS30ANTs', resolution=1, desc='4', suffix='dseg'); \
-               tfapi.get('OASIS30ANTs|NKI', resolution=1, label='brain', suffix='probseg'); \
-               tfapi.get('MNI152NLin2009cAsym|OASIS30ANTs|NKI', resolution=1, desc='BrainCerebellumRegistration', suffix='mask'); "
-
 # Installing dev requirements (packages that are not in pypi)
 WORKDIR /src/
 COPY requirements.txt requirements.txt
 RUN pip install --no-cache-dir -r requirements.txt && \
     rm -rf $HOME/.cache/pip
 
+# Precaching atlases
+RUN python -c "from templateflow import api as tfapi; \
+               tfapi.get(['MNI152Lin', 'MNI152NLin2009cAsym', 'OASIS30ANTs'], suffix='T1w'); \
+               tfapi.get(['MNI152Lin', 'MNI152NLin2009cAsym', 'OASIS30ANTs'], desc='brain', suffix='mask'); \
+               tfapi.get('OASIS30ANTs', resolution=1, desc='4', suffix='dseg'); \
+               tfapi.get(['OASIS30ANTs', 'NKI'], resolution=1, label='brain', suffix='probseg'); \
+               tfapi.get(['MNI152NLin2009cAsym', 'OASIS30ANTs', 'NKI'], resolution=1, desc='BrainCerebellumRegistration', suffix='mask'); "
```

Comment:
The surface issue appears to have been that the atlases were specified as regexes, not lists. The deeper issue is that templateflow is not being used to fetch these files at need.